apiserver/facades/agent/provisioner/provisioner.go

```diff
@@ -314,106 +313,30 @@ func (p *ProvisionerAPI) WatchAllContainers(args params.WatchContainers) (params
 	return p.WatchContainers(args)
 }
 
-// WatchContainersCharmProfiles starts a StringsWatcher to  notifies when
+// WatchContainersCharmProfiles starts a StringsWatcher to notify when
 // the provisioner should update the charm profiles used by a container on
 // the given machine.
+//
+// Remove in juju 3, this is for compatibility of provisioner using APIV7.
+// It's required for the worker to start, and for the former
+// processProfileChanges, which will no longer be used in APIv9.
+// Code in apiserver application SetCharm() will prevent APIv8
+// models from upgrading charm with profiles and causing issues.
```

Comment:
Because ProvisionerAPIV7 embeds ProvisionerAPIV8, you don't need to duplicate this WatchContainersCharmProfiles() method here on V7.